app/models/firm.rb

```diff
@@ -189,11 +189,7 @@ def main_office
   end
 
   def publishable?
-    registered? && offices.any? && !missing_advisers?
-  end
-
-  def missing_advisers?
-    (primary_advice_method == :local) && advisers.empty?
+    registered? && offices.any? && advisers.any?
```

Comment:
👏 

Just to make it clear: by changing the behaviour of the `publishable?` method, we also [control who's indexed or not (or removed from the existing index)](https://github.com/moneyadviceservice/rad/blob/0e797a10757914475efe858708cc2c207a001512/lib/firm_indexer.rb#L4-L8). That is specifically when:

1. The index is [manually (re)created](https://github.com/moneyadviceservice/rad/blob/6b9c0eca644b0d3fd9878b778cb4caf0f4fa7eef/lib/tasks/index.rake#L11) (we don't really do it apparently)
2. An update is triggered on a [firm](https://github.com/moneyadviceservice/rad/blob/0e797a10757914475efe858708cc2c207a001512/app/models/firm.rb#L113), [adviser](https://github.com/moneyadviceservice/rad/blob/0e797a10757914475efe858708cc2c207a001512/app/models/adviser.rb#L37) or [office](https://github.com/moneyadviceservice/rad/blob/0e797a10757914475efe858708cc2c207a001512/app/models/office.rb#L43) model (i.e. addition/update from the weekly FCA import _or_ manual update from the self signup area _or_ by a RAD admin) 

This should not be a problem, since it would still allow some firms/advisers to have a grace period.

I just want to highlight that: **some firms/advisers might start being removed from the index because of any of 2. happens at any point in time.** Better confirm that with product, still possibly fine and in line with what we want, but just to be extra clear about it.

Reply:
Thanks @your, I will confirm with product. The issue is that product are asking for conflicting functionality; the warning message is dependent on the `publishable?` method. 

I considered would be to use the `created_at` field to preclude firms created before a designated date from being removed from the index. This approach which would allow the onboarding warning to be displayed. But I would argue against that approach.

Reply:
@MarcoBgn I think it should be fine and we should keep it simple, basically they will all get removed _eventually_. It's just that we cannot predict exactly _when_ at the moment if we introduce these changes (which by the way, I do approve), because several events can cause the removal. As long as product is aware of it, maybe it's just fine. Hoping they have some comms ready for those firms/advisers in case.

Reply:
I have spoken to product and to the stakeholder who has confirmed this approach and will initiate the necessary comms to ensure the existing remote firms with 0 advisers update their listings. I believe this pr can progress to a test environment, as long as we don't progress to production until we get the go-ahead from the stakeholder.